### PR TITLE
chore: script to bump a module to version 1.0.0

### PR DESCRIPTION
**Pull request by suztomo**
Given an API shortname, this script bumps the corresponding module version to 1.0.0.

Usage:

`~/google-cloud-java` has the change in this pull request 10907 and `~/google-cloud-java-target` has the Release Please pull request https://github.com/googleapis/google-cloud-java/pull/10883.

```
suztomo@suztomo:~/google-cloud-java-target$ gh pr checkout 10883
branch 'release-please--branches--main' set up to track 'origin/release-please--branches--main'.
Switched to a new branch 'release-please--branches--main'
suztomo@suztomo:~/google-cloud-java-target$  ~/google-cloud-java/.github/ga_version_bump/bump_module_to_ga.sh securityposture
Key:google-cloud-securityposture, Value:1.0.0
Key:proto-google-cloud-securityposture-v1, Value:1.0.0
Key:grpc-google-cloud-securityposture-v1, Value:1.0.0
suztomo@suztomo:~/google-cloud-java-target$ git diff
diff --git a/gapic-libraries-bom/pom.xml b/gapic-libraries-bom/pom.xml
index 1cd29c59e37..ddd6e941289 100644
--- a/gapic-libraries-bom/pom.xml
+++ b/gapic-libraries-bom/pom.xml
@@ -1014,7 +1014,7 @@
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>google-cloud-securityposture-bom</artifactId>
-        <version>0.10.0</version><!-- {x-version-update:google-cloud-securityposture:current} -->
+        <version>1.0.0</version><!-- {x-version-update:google-cloud-securityposture:current} -->
         <type>pom</type>
         <scope>import</scope>
       </dependency>
diff --git a/java-securityposture/google-cloud-securityposture-bom/pom.xml b/java-securityposture/google-cloud-securityposture-bom/pom.xml
index d555042cc51..886b932d508 100644
--- a/java-securityposture/google-cloud-securityposture-bom/pom.xml
+++ b/java-securityposture/google-cloud-securityposture-bom/pom.xml
@@ -3,7 +3,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-securityposture-bom</artifactId>
-  <version>0.10.0</version><!-- {x-version-update:google-cloud-securityposture:current} -->
suztomo@suztomo:~/google-cloud-java-target$ git commit -a
[release-please--branches--main 458a24ad676] securityposture to 1.0.0
 7 files changed, 18 insertions(+), 18 deletions(-)
```

The commit was added to the Release Please pull request: https://github.com/googleapis/google-cloud-java/pull/10883/commits/458a24ad676e36bcb6d66ed5c75e998c3c1aad5b.

### CODE REVIEW

File: .github/ga_version_bump/bump_module_to_ga.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# java-accessapproval) in the google-cloud-java repository to 1.0.0.
+# You need to call this script from the root directory on the Release Please pull request branch;
+# otherwise the changes to the versions are overridden by the Release Please.
+
+set -eo pipefail
+set -x
+
+# Pattern to match the lines in versions.txt. Usually it's the API shortname for a module.
+# For example "securityposture" for the "java-securityposture" module.
+shortname=$1
```
Comment (JoeWang1127) [inline .github/ga_version_bump/bump_module_to_ga.sh:13]: could you change this parameter to `library_name`? `library_name` is guarantee unique.

File: .github/ga_version_bump/bump_module_to_ga.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+versions_file=versions.txt
+
+sed -i -E "s/^(.*${shortname}(-v.+)?):0.+:.+/\1:1.0.0:1.0.0/g" "${versions_file}"
+
+# Guardrail for bad pattern
+set +e
+diff_lines_count=$(git diff --unified=0 versions.txt |grep -c "^[+-]\w")
+
+if ! grep --quiet -E "${shortname}-v[0-9]:" versions.txt; then
```
Comment (JoeWang1127) [inline .github/ga_version_bump/bump_module_to_ga.sh:23]:
```suggestion
if ! grep --quiet -E "${shortname}-v[1-9]:" versions.txt; then
```

File: .github/ga_version_bump/bump_module_to_ga.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  exit 1
+fi
+
+set -e
+if [[ "${diff_lines_count}" -eq 0 ]]; then
+  echo "There's no matching lines in versions.txt for ${shortname}."
+  echo "Check the hyphenation in the versions.txt."
+  exit 1
+fi
+if [[ "${diff_lines_count}" -gt 10 ]]; then
```
Comment (JoeWang1127) [inline .github/ga_version_bump/bump_module_to_ga.sh:35]: We don't need this if statement if `library_name` is unique.

**Comment by JoeWang1127** [comment]
Could you paste the running result (changes in versions.txt) after running the script with a given shortname/library_name?

File: .github/ga_version_bump/bump_module_to_ga.sh (hunk 3 above)
Comment (suztomo) [reply .github/ga_version_bump/bump_module_to_ga.sh:35]: It's possible that the caller puts a bad value here such as "r".

**Comment by suztomo** [comment]
> Could you paste the running result 

Added the usage example in description with an example commit.

File: .github/ga_version_bump/bump_module_to_ga.sh (hunk 2 above)
Comment (suztomo) [reply .github/ga_version_bump/bump_module_to_ga.sh]: Fixed.

File: .github/ga_version_bump/bump_module_to_ga.sh (hunk 1 above)
Comment (suztomo) [reply .github/ga_version_bump/bump_module_to_ga.sh]: Updated.

**Review by diegomarquezp — APPROVED** [review]

**Pull request merged**